src/main/java/com/burgstaller/okhttp/AuthenticationCacheInterceptor.java

```diff
@@ -50,15 +46,28 @@ public Response intercept(Chain chain) throws IOException {
 
         // Cached response was used, but it produced unauthorized response (cache expired).
         int responseCode = response != null ? response.code() : 0;
-        if (authenticator != null && (responseCode == HTTP_UNAUTHORIZED || responseCode == HTTP_PROXY_AUTH)) {
-            // Remove cached authenticator and resend request
-            if (authCache.remove(key) != null) {
-                response.body().close();
-                Platform.get().log("Cached authentication expired. Sending a new request.", Platform.INFO, null);
-                // Force sending a new request without "Authorization" header
+
+        if (authenticator != null && (
+                (cacheKeyProvider.applyToProxy() && responseCode == HTTP_PROXY_AUTH) ||
+                        !cacheKeyProvider.applyToProxy() && responseCode == HTTP_UNAUTHORIZED)
+        ) {
+            response = removeCacheEntry(chain, key, request, response);
+        }
+        return response;
+    }
+
+    private Response removeCacheEntry(Chain chain, String key, Request request, Response response) throws IOException {
```

Comment:
I would appreciate if you can change the method name to make it clearer what it does. It not only removes the cache entry but also sends the a new request.

Reply:
based on your feedback, i've changed my mind : it seems more clear to do one thing in a method.
Like there is 2 different ways, i've removed this new method, and integrated these 2 different ways in the current calling method.